platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -16,46 +16,97 @@
+            table.addRow(
+                    "address,",
+                    address.getNodeId() + ",",
+                    address.getNickname() + ",",
+                    address.getSelfName() + ",",
+                    address.getWeight() + ",",
+                    (hasInternalIpv4 ? ipString(address.getAddressInternalIpv4()) : "") + ",",
+                    address.getPortInternalIpv4() + ",",
+                    (hasExternalIpv4 ? ipString(address.getAddressExternalIpv4()) : "") + ",",
+                    address.getPortExternalIpv4() + (hasMemo ? "," : ""),
+                    memo);
+        }
+        return table.render();
+    }
+
+    /**
+     * Parses an address book from text in the form described by config.txt.  Comments are ignored.
+     *
+     * @param addressBookText the config.txt compatible serialized address book to parse.
+     * @return a parsed AddressBook.
+     * @throws ParseException if any Address throws a ParseException when being parsed.
+     */
+    @NonNull
+    public static AddressBook parseAddressBookText(@NonNull final String addressBookText) throws ParseException {
+        Objects.requireNonNull(addressBookText, "The addressBookText must not be null.");
+        final AddressBook addressBook = new AddressBook();
+        for (final String addressLine : addressBookText.split("\\r?\\n")) {
+            final Address address = parseAddressText(addressLine);
+            if (address != null) {
+                addressBook.add(address);
+            }
+        }
+        return addressBook;
+    }
+
+    /**
+     * Parse an address from a single line of text, if it exists.  Address lines may have comments which start with the
+     * `#` character.  Comments are ignored.  Lines which are just comments return null.  If there is content prior to a
+     * `#` character, parsing the address is attempted.  Any failure to generate an address will result in throwing a
+     * parse exception.  The address parts are comma separated.   The format of text addresses prevent the use of `#`
+     * and `,` characters in any of the text based fields, including the memo field.
+     *
+     * @param addressText the text to parse.
+     * @return the parsed address or null if the line is a comment.
+     * @throws ParseException if there is any problem with parsing the address.
+     */
+    @Nullable
+    public static Address parseAddressText(@NonNull final String addressText) throws ParseException {
+        Objects.requireNonNull(addressText, "The addressText must not be null.");
+        // lines may have comments which start with the first # character.
+        final String[] textAndComment = addressText.split("#");
+        if (textAndComment.length == 0
+                || textAndComment[0] == null
+                || textAndComment[0].trim().isEmpty()) {
+            return null;
+        }
+        final String[] parts = addressText.split(",");
```

Comment:
ok, from my point of view it really makes sense to define how `addressText` is structured. Otherwise you always need to do reverse eng to understand the code ;)

Reply:
I will add more comments.